src/main/java/edu/ie3/datamodel/utils/validation/ThermalValidationUtils.java

```diff
@@ -154,10 +160,13 @@ private ThermalValidationUtils() {
     List<Try<Void, ? extends ValidationException>> exceptions = new ArrayList<>();
 
     // Further checks for subclasses
-    if (CylindricalStorageInput.class.isAssignableFrom(thermalStorageInput.getClass())) {
+    if (thermalStorageInput.getClass() == CylindricalStorageInput.class) {
```

Comment:
Why did you change this? Couldn't you just switch the order of the inputs to check?